tenable/ot/graphql/definitions.py

```diff
@@ -63,77 +65,105 @@ def __str__(self):
         return self.message
 
 
-class LocationSchema(Schema):
-    '''
-    Schema for GraphQL location part of the error.
-    '''
-    line = fields.Int(required=True)
-    column = fields.Int(required=True)
+class SchemaBase(Schema):
+    dataclass = None
+    snake_case_regex1 = re.compile("(.)([A-Z][a-z]+)")
+    snake_case_regex2 = re.compile("([a-z0-9])([A-Z])")
 
-    @post_load
-    def to_object(self, data, **kwargs):
-        '''This method turns the schema into its corresponding object.'''
-        return Location(**data)
+    def __init__(self, *args, **kwargs):
+        super(SchemaBase, self).__init__(*args, **kwargs)
 
-
-class ExtensionsSchema(Schema):
-    '''
-    Schema for GraphQL extensions part of the error.
-    '''
-    code = fields.Str()
+    def camel_to_snake(self, name):
+        name = self.snake_case_regex1.sub(r"\1_\2", name)
+        return self.snake_case_regex2.sub(r"\1_\2", name).lower()
 
     @post_load
-    def to_object(self, data, **kwargs):
-        '''This method turns the schema into its corresponding object.'''
-        return Extensions(**data)
+    def to_object(self, data, *args, **kwargs):
+        """This method turns the schema into its corresponding object."""
+        if self.dataclass is None:
+            raise ValueError(f"dataclass must be set for {type(self)}")
+        normalized_data = {
```

Comment:
not sure if .item() can failed, but maybe check that data is dict?

if type(data) is not dict:
      raise ValueError(f"data should be from type dict")

Reply:
done